dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.ts

```diff
@@ -167,6 +173,20 @@ export class TfmDealCancellationRepo {
       throw new DealNotFoundError(dealId.toString());
     }
 
-    return { cancelledDealUkefId: dealId };
+    const facilityCollection = await this.getFacilityCollection();
+
+    await facilityCollection.updateMany(
+      { 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } },
+      flatten({
+        'tfm.facilityStage': TFM_FACILITY_STAGE.RISK_EXPIRED,
+        auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails),
+      }),
+    );
+
+    const updatedFacilities = await facilityCollection.find({ 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } }).toArray();
+
+    const updatedFacilityUkefIds = updatedFacilities.map((facility) => facility.facilitySnapshot.ukefFacilityId).filter((id) => id !== null);
```

Comment:
I think we may already have a helper for this @MarRobSoftwire created in another PR, can we use this?

Reply:
That's in tfm-api currently but it will be removed once this code is in, I think makes sense to copy the helper for now & remove the current one later

Reply:
Thanks @MarRobSoftwire  - can/should we create a subtask for this so we don't forget?

Reply:
Thanks both! If OK with you @ttbarnes now @MarRobSoftwire's latest PR is merged in, I'm happy to do another commit here, moving the helper to dtfs-central and removing it from tfm-api (and updating the logic in tfm-api respectively). Will add a few more file changes to the diff here, and need a re-review but I think it will be the neatest option!

Reply:
Sounds good to me @BethThomas141 👍